src/Weasel.Postgresql/PostgresqlDatabase.cs

```diff
@@ -7,6 +8,12 @@ namespace Weasel.Postgresql;
 
 public abstract class PostgresqlDatabase: DatabaseBase<NpgsqlConnection>
 {
+    /// <summary>
+    /// Provide status of whether the Npgsql types were reloaded
+    /// Currently used for assertion in unit tests
+    /// </summary>
+    public bool HasNpgsqlTypesReloaded { get; private set; }
```

Comment:
**SUGGESTION:** I don't think that we should expose public property just for the sake of tests. I think that we should install in the test project some extension that requires a type reload and ensure that it's actually working. The test just checks if the code was run. In the worst case, we should make this property internal and allow access for test project to internal code, still, I'd treat this as a last resort if we cannot easily run tests against extensions.

Reply:
> I think that we should install in the test project some extension that requires a type reload and ensure that it's actually working.
May be I can look at this then we don't need that property.

Reply:
Quick update:
- I tried with citext and Npgsql looks to be loading the types by default so it is working fine.
- Further went and tried to test with PostGis in my local, just refreshing types didn't work to fetch data of say of type point or geometry. Code something like below:

```csharp
[Fact]
    public async Task test_auto_reload_of_npgsql_types()
    {
        var theDatabase = TestDatabase.ForConnectionString(ConnectionSource.ConnectionString);
        theDatabase.ExtensionFeatures["Extensions"].AddExtension("postgis");
        await theDatabase.ApplyAllConfiguredChangesToDatabaseAsync(AutoCreate.CreateOrUpdate);
        var conn = this.theDatabase.CreateConnection();
        await conn.OpenAsync();
        var cmd = conn.CreateCommand("select $1");
        cmd.Parameters.Add(new() { Value = new Point(new Coordinate(1d, 2d)) });
        var reader = await cmd.ExecuteReaderAsync();
        await reader.ReadAsync();
        var val = reader.GetFieldValue<Point>(0);
    }
```

Getting an exception as below:

```
System.InvalidCastException
Writing values of 'NetTopologySuite.Geometries.Point' is not supported for parameters having no NpgsqlDbType or DataTypeName. Try setting one of these values to the expected database type..
   at Npgsql.Internal.AdoSerializerHelpers.<GetTypeInfoForWriting>g__ThrowWritingNotSupported|1_0(Type type, PgSerializerOptions options, Nullable`1 pgTypeId, Nullable`1 npgsqlDbType, Exception inner)
   at Npgsql.Internal.AdoSerializerHelpers.GetTypeInfoForWriting(Type type, Nullable`1 pgTypeId, PgSerializerOptions options, Nullable`1 npgsqlDbType)
   at Npgsql.NpgsqlParameter.ResolveTypeInfo(PgSerializerOptions options)
   at Npgsql.NpgsqlParameterCollection.ProcessParameters(PgSerializerOptions options, Boolean validateValues, CommandType commandType)
   at Npgsql.NpgsqlCommand.ExecuteReader(Boolean async, CommandBehavior behavior, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(Boolean async, CommandBehavior behavior, CancellationToken cancellationToken)
   at Weasel.Postgresql.Tests.Migrations.SchemaMigrationTests.test_auto_reload_of_npgsql_types() in /Users/babuannamalai/oss_contrib/weasel_main_repo/src/Weasel.Postgresql.Tests/Migrations/migration_scenario_tests.cs:line 134
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_0.<<InvokeTestMethodAsync>b__1>d.MoveNext() in /_/src/xunit.execution/Sdk/Frameworks/Runners/TestInvoker.cs:line 285
--- End of stack trace from previous location ---
   at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in /_/src/xunit.execution/Sdk/Frameworks/ExecutionTimer.cs:line 48
   at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in /_/src/xunit.core/Sdk/ExceptionAggregator.cs:line 90
```

In summary, there looks to be more this than just refresh the Npgsql types. I will investigate further on this.